### ISSUE TITLE
Validation check failing

### ISSUE THREAD

**Issue by mikeraynham** [ffrank/puppet-constraints#1]
Hi,

I've been attempting to use this module with Puppet version 3.7.2, but have encountered an issue with [this check](https://github.com/ffrank/puppet-constraints/blob/cf72e86b3e024a2959627c89b8ff54f437486c49/lib/puppet/type/constraint.rb#L157-L158) failing.  I've tried my own code, and I've copied the examples from the documentation, but it appears as though at this stage, the resources are just strings.  For debugging purposes I tried this (please forgive my first attempt at Ruby):

``` ruby
types = self[:resource].map { |res| [res.class, res.to_s].join(': ') }.join(', ')
fail "resource must be a(n array of) resource reference(s): " + types \
    if ! self[:resource].select { |res| ! res.is_a? Puppet::Resource }.empty?
```

Given:

``` ruby
constraint { 'core_dependencies':
    resource => [ Package['clamav'], Package['clamd'] ],
    allow    => { ensure => present };
}
```

...the error is:

```
Error: Failed to apply catalog: Validation of Constraint[core_dependencies] failed: resource must be a(n array of) resource reference(s): String: Package[clamav], String: Package[clamd] at /etc/puppet/modules/sr_app/manifests/install/core_dependencies.pp:42
Wrapped exception:
resource must be a(n array of) resource reference(s): String: Package[clamav], String: Package[clamd]
```

I've tried removing the test, after which the module appears to work correctly if I only specify one resource.  However, if I specify multiple resources, it fails with a constraint error:

```
Error: Got an uncaught exception of type RuntimeError: the resource Package[clamav]Package[clamd] cannot be found in the catalog
```

**Comment by ffrank** [comment]
Hi there,

sorry I missed the mail about this completely. Since November. Wow.

I'll look into it asap. Cheers!

**Comment by ffrank** [comment]
Interesting: This bug only manifests itself when using master/agent.

Given the following file `/tmp/const-test.pp`

```
constraint { 'core_dependencies':
    resource => [ Package['clamav'], Package['clamd'] ],
    allow    => { ensure => present };
}

package {
    'clamav': ensure => present;        
    'clamd':  ensure => present;
}
```

The `apply` just works:

```
$ bundle exec puppet apply /tmp/const-test.pp --noop
Notice: Compiled catalog for geras.fritz.box in environment production in 0.45 seconds
Notice: /Stage[main]/Main/Package[clamd]/ensure: current_value purged, should be present (noop)
Notice: /Stage[main]/Main/Package[clamav]/ensure: current_value purged, should be present (noop)
Notice: Class[Main]: Would have triggered 'refresh' from 2 events
Notice: Stage[main]: Would have triggered 'refresh' from 1 events
Notice: Finished catalog run in 0.85 seconds
```

With the same code in `site.pp`, the agent barfs as reported.

**Comment by ffrank** [comment]
@mikeraynham I pushed a [fix](https://github.com/ffrank/puppet-constraints/commit/a8a93b74f8bfb223d6d8720ddaf8412b3c9c9f92) to master. It's pending release until I find a good way to write a test for this.

Can you try and see if this fixes your issue? It should not be difficult to patch your local copy.

Thanks for reporting! Sorry for the huge delay.

**Comment by ffrank** [comment]
I just released version 0.2.1. I believe that this issue is now fixed.

Thanks again for reporting @mikeraynham !

**Comment by mikeraynham** [comment]
@ffrank I've applied this update, and it works perfectly.  Thank you.
